### PR TITLE
docs: QE migration guide

### PR DESCRIPTION
**Pull request by alexjuda**
# The problem

QE is going to be sunset soon. We need a reliable source for migrating workflows to use CE.

# This PR's solution

A dedicated Workflow SDK guide.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: docs/examples/tests/test_migrating_from_quantum_engine.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+
+def _import_models(
+    dsl_task,
+) -> t.Tuple[ir.Import, t.Optional[t.Sequence[ir.Import]]]:
+    """
+    Uses the ``dsl_task`` exactly once in a workflow, builds the workflow model, and
+    extracts imports used by the corresponding task def model.
+    """
+    # TODO: reuse across test files
```
Comment (SebastianMorawiec) [inline docs/examples/tests/test_migrating_from_quantum_engine.py:131]: Add a ticket to our backlog and ref it here in TODO

File: docs/examples/tests/test_migrating_from_quantum_engine.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    src_import = wf_model.imports[task_model.source_import_id]
+    if (deps_ids := task_model.dependency_import_ids) is not None:
+        deps_imports = [wf_model.imports[dep_id] for dep_id in deps_ids]
+    else:
+        deps_imports = None
+
+    return src_import, deps_imports
+
+
+@pytest.mark.slow
```
Comment (SebastianMorawiec) [inline docs/examples/tests/test_migrating_from_quantum_engine.py:149]: Question, why are those tests slow? I dont see them using like Ray or running any workflows in particular

File: docs/guides/migrating-from-quantum-engine.rst [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+By August 2023, Quantum Engine (QE) is being sun set.
+Running workflows remotely will only work with Compute Engine (CE).
+This document covers the differences between QE and CE, and how to move to CE.
+
+
+Am I Affected?
+==============
+
+If you're using ``orq login`` or ``orq login --qe``, you're on QE. Read this guide.
```
Comment (SebastianMorawiec) [inline docs/guides/migrating-from-quantum-engine.rst:13]: Question: Isnt orq login default to CE since like 2 releases already?

**Review by SebastianMorawiec — APPROVED** [review]
Praise: Great doc! nice work
Thought: Im not sure if tests for QE code was required, a bit overkill imho but as they are already there.... lets leave them :)

File: docs/examples/tests/test_migrating_from_quantum_engine.py (hunk 2 above)
Comment (alexjuda) [reply docs/examples/tests/test_migrating_from_quantum_engine.py]: Thanks, fixed

File: docs/guides/migrating-from-quantum-engine.rst (hunk 3 above)
Comment (alexjuda) [reply docs/guides/migrating-from-quantum-engine.rst]: Thanks, fixed

**Pull request merged**